docs/source/guides/metrics.rst

```diff
@@ -0,0 +1,74 @@
+=========================
+Custom Metrics in BentoML
+=========================
+Metrics are used for monitoring different aspect of user-defined APIs. 
+In concept, being able to closely monitor how endpoints are behaving, and being used, will give you more control over how to customize and optimization your endpoints. 
+In practice, ``bentoml.metrics`` supports all exposed functions from ``prometheus_client``, which in turn means that you can create any custome metric following the ``prometheus_client`` standards.
+Furthermore, by replacing ``prometheus_client`` with ``bentoml.metrics``, user can use all of the features ``prometheus_client`` offers. 
+
```

Comment:
Add a tutorial on how to add a histogram and query it with PromQL. 

Add a tutorial for custom runner using one of the examples.